spa/src/components/pageEditor/elementEditors/amount/AmountEditor.tsx

```diff
@@ -9,10 +9,38 @@ import { ElementDetailEditorProps } from 'components/pageEditor/editInterface/El
 export type AmountEditorProps = ElementDetailEditorProps<AmountElement['content']>;
 
 export function AmountEditor({ contributionIntervals, elementContent, onChangeElementContent }: AmountEditorProps) {
-  function handleAddAmount(interval: ContributionInterval, value: number) {
-    // Keep the list of amounts sorted numerically.
+  useEffect(() => {
+    // Migrate legacy data to the new format on first render.
```

Comment:
```suggestion
    // Migrate legacy data to the new format.
```
The effect will run whenever `elementContent` changes, so this comment's a little misleading.